tiledb/sm/cpp_api/map.h

```diff
@@ -36,17 +36,488 @@
 
 #include "context.h"
 #include "deleter.h"
-#include "map_item.h"
-#include "map_iter.h"
 #include "map_schema.h"
 #include "utils.h"
 
 #include <map>
 #include <memory>
 #include <string>
 
+// If the user doesn't provide a key, use this key
+#define TILEDB_SINGLE_ATTRIBUTE_MAP "value"
```

Comment:
Can we `#undef` this at the end of this file?

Reply:
I removed this define completely, the user needs to define an attribute name when constructing a `tiledb::Map` from a `std::[unordered_]map` now.